centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1899,24 +1901,17 @@ objDataToWords(int data_type, void* data_ptr)
 	else if (data_type == DATA_T_MONEY)
 	    {
 	    m = (pMoneyType)data_ptr;
-	    if (m->WholePart < 0)
+	    if (m->Value < 0)
 	        {
-		if (m->FractionPart == 0)
-		    {
-		    integer_part = -m->WholePart;
-		    fraction_part = 0;
-		    }
-		else
-		    {
-		    integer_part = (-m->WholePart) - 1;
-		    fraction_part = 10000 - m->FractionPart;
-		    }
+	        /** Keep the opposite (positive) value for printing **/
+            integer_part = -m->Value/10000;
+            fraction_part = -m->Value%10000;
 		xsConcatenate(&tmpbuf, "Negative ", -1);
 		}
 	    else
 	        {
-		integer_part = m->WholePart;
-		fraction_part = m->FractionPart;
+		integer_part = m->Value/10000;
+		fraction_part = m->Value%10000;
```

Comment:
As in, the fraction part was potentially larger than m->Value%10000?

Reply:
The old representation defined the whole part as floor(value) and the fraction part as value - floor(value). This is not the same as using % to get the remainder when you're working with negative values. With the original representation, -$0.0001 would be represented as { -1, 9999 }, not { 0, -1 }, i.e., the fraction part always moves the value in the positive direction, rather than away from zero.

Reply:
Right.  Upon review of the function in the old and the new, doesn't  m->Value%10000 return the proper value for FractionPart, even though it is not equivalent to the old representation?  The old representation set "fraction_part = 10000 - m->FractionPart;", and 10000-m->FractionPart is the same as m->Value%10000.  I think this was a function where I did not change the logic but tried to format the same inputs.

Also a note, objDataToWords() truncates past normal cent value for the string representation, so if that is being used to display how much money someone has, there might be discrepancies between the surface representation and what is actually stored, since we have decided upon rounding for the logic

Reply:
Good point, Carl, the above two lines of code only run when the value is positive.